src/placeos-rest-api/controllers/mqtt.cr

```diff
@@ -0,0 +1,119 @@
+require "./application"
+
+module PlaceOS::Api
+  class MQTT < Application
+    base "/api/engine/v2/mqtt/"
+
+    # Params
+    ###############################################################################################
+
+    getter mqtt_client_id_param : String? do
+      params["clientid"]?
+    end
+
+    getter mqtt_topic_parma : String? do
+      params["topic"]?
+    end
+
+    getter mqtt_access_param : Int32? do
+      params["acc"]?.try(&.to_i?)
+    end
+
+    ###############################################################################################
+
+    # For MQTT JWT access: https://github.com/iegomez/mosquitto-go-auth#remote-mode
+    # jwt_response_mode: status, jwt_params_mode: form
+    post "/mqtt_user", :mqtt_user do
```

Comment:
Since this is already scoped beneath `mqtt`, can we remove the prefix?
```suggestion
    post "/user", :mqtt_user do
```

Reply:
I was thinking of keeping the prefix as it's not technically an API that we'll publish to anyone and it makes the filters clearer

```crystal
    # All routes are authenticated, except root
    before_action :authorize!, except: [:root, :mqtt_user, :mqtt_access]

    # Simplifies determining user's requests in server-side logs
    before_action :set_user_id, except: [:root, :mqtt_user, :mqtt_access]
```

https://github.com/PlaceOS/rest-api/blob/mqtt-permissions-check/src/placeos-rest-api/controllers/application.cr#L78

Reply:
Those filters can keep the prefix IMO, I just think it's a bit neater given that the route is already nested under the MQTT namespace

Reply:
If anything, they should be `/mqtt/mosquitto_access` and `/mqtt/mosquitto_user` because I believe this is not standard behaviour, correct me if I'm wrong